src/autorest-core/lib/message.ts

```diff
@@ -30,7 +30,10 @@ export interface Message {
   Key?: Iterable<string>;
   Details?: any;
   Text: string;
+
+  // injected or modified by core
   Source?: Array<SourceLocation>;
   Range?: Iterable<Range>;
   Plugin?: string;
+  FormattedMessage?: string;
```

Comment:
cool, feel free to start emitting such properties - autorest does not throw away properties on messages anyway, so adding it to `Details` from, say, the validator plugin will work out of the box (you'll see it in CI, in the VS code extension, etc.).